lib/spack/spack/database.py

```diff
@@ -1526,15 +1526,15 @@ def get_by_hash(self, dag_hash, default=None, installed=any):
     def _query(
         self,
         query_spec=any,
-        known=any,
+        selection_fn: SelectType = None,
```

Comment:
maybe call it predicate?

Reply:
I don't mind calling it predicate, but we have many other places where callables that can be passed as argument are suffixed `_fn`, so I was using that suffix for consistency.

Reply:
Then I would call it predicate_fn ;) selection is a bit vague

Reply:
Done in https://github.com/spack/spack/pull/47102/commits/23356d6d413a3f92af1fa0e0d25ab1244098d23d